libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/DefaultTracer.java

```diff
@@ -44,13 +45,13 @@ public SpanScope startSpan(String spanName) {
 
     @Override
     public SpanScope startSpan(String spanName, Attributes attributes) {
-        return startSpan(spanName, null, attributes);
+        return startSpan(spanName, SpanContext.EMPTY, attributes);
     }
 
     @Override
     public SpanScope startSpan(String spanName, SpanContext parentSpan, Attributes attributes) {
         Span span = null;
-        if (parentSpan != null) {
+        if (parentSpan != null && parentSpan.getSpan() != null) {
```

Comment:
Please remove the `parentSpan.getSpan() != null` - there should no possibility to create such  `SpanContext` instances